ipn/ipnlocal/network-lock.go

```diff
@@ -559,10 +552,32 @@ func (b *LocalBackend) NetworkLockStatus() *ipnstate.NetworkLockStatus {
 		NodeKeySignature: nodeKeySignature,
 		TrustedKeys:      outKeys,
 		FilteredPeers:    filtered,
+		VisiblePeers:     visible,
 		StateID:          stateID1,
 	}
 }
 
+func tkaStateFromPeer(p tailcfg.NodeView) ipnstate.TKAPeer {
+	fp := ipnstate.TKAPeer{
+		Name:         p.Name(),
+		ID:           p.ID(),
+		StableID:     p.StableID(),
+		TailscaleIPs: make([]netip.Addr, p.Addresses().Len()),
+		NodeKey:      p.Key(),
+	}
+	for i := range p.Addresses().Len() {
+		addr := p.Addresses().At(i)
+		if addr.IsSingleIP() && tsaddr.IsTailscaleIP(addr.Addr()) {
+			fp.TailscaleIPs[i] = addr.Addr()
+		}
```

Comment:
if `p.Addresses` contains non-tailscale IPs, `fp.TailscaleIPs` will have a few empty items at the end of the slice.
might be safer to use `append` here and use `p.Addresses().Len()` as pre-allocated capacity.

Reply:
Good point, thanks! Changed.